src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -106,7 +106,7 @@ public IActionResult OnPost(string provider, string returnUrl = null)
                 var info = await _signInManager.GetExternalLoginInfoAsync();
                 if (info == null)
                 {
-                    throw new ApplicationException("Error loading external login information during confirmation.");
+                    return NotFound("Error loading external login information during confirmation.");
```

Comment:
Should probably be InvalidOperation because it doesn't look to me that this is based on the info from the request. It looks like this error condition is caused by other data in the state of the request.

Reply:
Fair enough. Thanks!